### PR TITLE
Skip running effects so compononts with useLayoutEffect can be correctly prepassed

### PR DESCRIPTION
**Pull request by BartWaardenburg**
Thanks to @JoviDeCroock !

Fixes: https://github.com/sventschui/preact-ssr-prepass/issues/4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
   },
   "devDependencies": {
     "@babel/core": "^7.4.5",
+    "@babel/helper-compilation-targets": "^7.10.4",
```
Comment (JoviDeCroock) [inline package.json:37]: Why is this needed?

Comment (BartWaardenburg) [reply package.json:37]: Not having that installed gave me this error: 🤷‍♂️ 
image

File: src/index.js [hunk 2]
```diff
@@ -20,108 +20,97 @@ type Options = {
...
-  }
-
-  if (props && getChildren((children = []), props.children).length) {
-    return Promise.all(
-      children.map((child) => prepass(child, visitor, context))
-    );
-  }
-
-  return Promise.resolve();
+	vnode/*: VNode */,
```
Comment (JoviDeCroock) [inline src/index.js]: I think your prettier ran here 😅

File: package.json (hunk 1 above)
Comment (JoviDeCroock) [reply package.json:37]: Ah yeah you got a newer node version so it doesn't work, you need the newest babel version 😅 I think just upgrading the `@babel` deps to 7.10.4 should be sufficient

File: src/index.js [hunk 3]
```diff
@@ -39,6 +39,8 @@ export default function prepass(
 		let c = vnode.__c = new Component(props, context);
 		c.__v = vnode;
 
+		options._skipEffects = options.__s = true;
```
Comment (JoviDeCroock) [inline src/index.js:42]: Do we want to reset this after the render has run to its original value/undefined? Wondering mainly because of other processes but I don't think it's going to introduce breaking stuff, should not carry over but let's do it for safety
